ledger/ledger-api-bench-tool/src/test/suite/scala/com/daml/ledger/api/benchtool/submission/NonTransientContractsITSpec.scala

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.benchtool.submission
+
+import com.daml.ledger.api.benchtool.config.WorkflowConfig
+import com.daml.ledger.api.benchtool.services.LedgerApiServices
+import com.daml.ledger.api.benchtool.BenchtoolSandboxFixture
+import com.daml.ledger.api.testing.utils.SuiteResourceManagementAroundAll
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.client.binding
+import com.daml.scalautil.Statement.discard
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{AppendedClues, Checkpoints, EitherValues, OptionValues}
+
+import scala.concurrent.Future
+
+class NonTransientContractsITSpec
+    extends AsyncFlatSpec
+    with BenchtoolSandboxFixture
+    with SuiteResourceManagementAroundAll
+    with Matchers
+    with AppendedClues
+    with EitherValues
+    with OptionValues
+    with Checkpoints {
+
+  it should "populate participant with contracts using specified application-ids and submitters" in {
+    val submissionConfig = WorkflowConfig.FooSubmissionConfig(
+      numberOfInstances = 100,
+      numberOfObservers = 1,
+      uniqueParties = false,
+      instanceDistribution = List(
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo1",
+          weight = 1,
+          payloadSizeBytes = 0,
+        ),
+        WorkflowConfig.FooSubmissionConfig.ContractDescription(
+          template = "Foo2",
+          weight = 1,
+          payloadSizeBytes = 0,
+        ),
+      ),
+      consumingExercises = Some(
+        WorkflowConfig.FooSubmissionConfig.ConsumingExercises(
+          probability = 0.7,
+          payloadSizeBytes = 0,
+        )
+      ),
+    )
+    for {
+      (apiServices, allocatedParties, submission) <- benchtoolFooSubmissionFixture(submissionConfig)
+      _ <- submission.performSubmission(submissionConfig)
+      txEvents: ObservedEvents <- txTreeObserver(
+        apiServices = apiServices,
+        party = allocatedParties.observers(0),
+      )
+    } yield {
+      val createAndConsumeOffsetPairs = for {
+        create <- txEvents.createEvents
+        consume <- txEvents.consumingExercises.find(_.contractId == create.contractId).toList
+      } yield create.offset -> consume.offset
+      val activeContracts = txEvents.createEvents.count(create =>
+        !txEvents.consumingExercises.exists(_.contractId == create.contractId)
+      )
+      val cp = new Checkpoint
+      val nonTransientContracts = createAndConsumeOffsetPairs.count {
+        case (createOffset, archiveOffset) => createOffset != archiveOffset
+      }
+      val transientContracts = createAndConsumeOffsetPairs.count {
+        case (createOffset, archiveOffset) => createOffset == archiveOffset
+      }
+      cp(discard(nonTransientContracts shouldBe 53))
+      cp(discard(transientContracts shouldBe 6))
+      cp(discard(activeContracts shouldBe 100 - 53 - 6))
```

Comment:
Adding a `totalContracts = 100` then using that above and here would make the calculation easier.  I think we need a way to flag a value as `randonSeedDependent`

Reply:
Refactored